extension/src/triggers.rs

```diff
@@ -0,0 +1,56 @@
+use pgx::*;
+
+#[pg_extern]
+fn timescale_analytics_probe() {
+
+}
+
+extension_sql!(r#"
+CREATE OR REPLACE FUNCTION disallow_experimental_dependencies()
+  RETURNS event_trigger
+ LANGUAGE plpgsql
+  AS $$
+DECLARE
+  guc_set TEXT;
+  cmd record;
+  experimental_schema_id oid;
+BEGIN
+
+  guc_set := current_setting('timescale_analytics_install_experimental', true);
+  IF guc_set IS NOT NULL AND guc_set = 'on' THEN
+    RETURN;
+  END IF;
+
+  SELECT oid schema_oid
+  INTO experimental_schema_id
+  FROM pg_catalog.pg_namespace
+  WHERE nspname='timescale_analytics_experimental'
+  LIMIT 1;
+
+  FOR cmd IN
+    SELECT top_dep.objid, top_dep.deptype, top_dep.refobjid
+    FROM pg_catalog.pg_depend top_dep
+    WHERE top_dep.refobjid=experimental_schema_id
+    AND top_dep.objid IN (
+      SELECT depend.refobjid dep_id
+      FROM pg_catalog.pg_depend depend
+      INNER JOIN (
+        SELECT obj.objid id
+        FROM pg_event_trigger_ddl_commands() as obj
+        WHERE NOT obj.in_extension
```

Comment:
Doesn't seem like there's a good thing for this, but I kinda wish we could exclude temporary objects from this as well.

Reply:
hmm, I don't care about that for this release, but maybe for a future version